hydromt/raster.py

```diff
@@ -1652,6 +1655,217 @@ def interpolate_na(self, method: str = "nearest", **kwargs):
+                # temp = self._obj.isel({self.y_dim:slice(u,u+h), self.x_dim: slice(l,l+w)})
+                if zl != 0:
+                    dst_transform = temp.raster.transform * temp.raster.transform.scale(
+                        2**zl
+                    )
+                    temp = temp.raster.reproject(
+                        dst_transform=dst_transform,
+                        dst_width=int(w / (2**zl)),
+                        dst_height=int(h / (2**zl)),
+                        method="average",
+                        # dst_crs=self.crs,
+                    )
+                if fmt == "tif":
+                    temp.raster.to_raster(f"{ssd}\\{row}.tif", driver="GTiff")
+                elif fmt == "nc":
+                    temp = temp.raster.gdal_compliant()
+                    temp.to_netcdf(f"{ssd}\\{row}.nc", engine="netcdf4")
+                else:
+                    raise ValueError(f"Ouput format (fmt) should be 'tif' or 'nc' -> {fmt} given")
+                
+                file.write(f"{ssd}\\{row}.{fmt}\n")
+
+                del temp
+
+            file.close()
+            # Create a vrt using GDAL
+            gis_utils.create_vrt(sd, mName)
+        # Write a quick yaml for the database
+        with open(f"{root}\\..\\{mName}.yml", "w") as w:
+            w.write(f"{mName}:\n")
+            crs = self.crs.to_epsg()
+            w.write(f"  crs: {crs}\n")
+            w.write("  data_type: RasterDataset\n")
+            w.write("  driver: raster\n")
+            w.write(f"  path: {mName}/{{zoom_level}}/{mName}.vrt\n")
+
+    def to_osm(
+        self,
+        root: str,
+        zl: int,
+        bbox: tuple = (),
+    ):
+        """Generate tiles from raster according to the osm scheme
+
+        Parameters
+        ----------
+        root : str
+            Path to folder where the database will be created
+        zl : int
+            Maximum zoom level of the database
+            Everyting is generated incrementally up until this level
+            E.g. zl = 8, levels generated: 0 to 7
+        bbox : tuple, optional
+            Bounding Box in the objects crs
+
+        """
+
+        assert self._obj.ndim == 2, "Only 2d datasets are accepted..."
+        obj = self._obj.transpose(self.y_dim, self.x_dim)
+
+        mName = os.path.normpath(os.path.basename(root))
+
+        def folder(path):
+            if not os.path.exists(path):
+                os.makedirs(path)
+
+        def transform_res(dres, transformer):
+            return transformer.transform(0, dres)[0]
+
+        folder(root)
+
+        dres = abs(self._obj.raster.res[0])
+        if bbox:
+            minx, miny, maxx, maxy = bbox
+        else:
+            minx, miny, maxx, maxy = self._obj.raster.transform_bounds(
+                dst_crs=self._obj.raster.crs
+            )
+
+        transformer = pyproj.Transformer.from_crs(self._obj.raster.crs.to_epsg(), 3857)
```

Comment:
could you add a small comment here what is done in this section

Reply:
OSM is not in this release anymore